pkg/controller/remotemachineset/remotemachineset_controller.go

```diff
@@ -352,65 +487,225 @@ func (r *ReconcileRemoteMachineSet) syncMachineSets(
+			}
 
-	for _, ms := range machineSets {
-		if ms.Labels == nil {
-			ms.Labels = map[string]string{}
 		}
-		ms.Labels[machinePoolNameLabel] = pool.Spec.Name
+	}
 
-		// Apply hive MachinePool labels to MachineSet MachineSpec.
-		ms.Spec.Template.Spec.ObjectMeta.Labels = make(map[string]string, len(pool.Spec.Labels))
-		for key, value := range pool.Spec.Labels {
-			ms.Spec.Template.Spec.ObjectMeta.Labels[key] = value
+	// Find MachineAutoscalers that need deleting
+	for i, rMA := range remoteMachineAutoscalers.Items {
+		if !isControlledByMachinePool(cd, pool, &rMA) {
+			continue
+		}
+		delete := true
+		if pool.DeletionTimestamp == nil && pool.Spec.Autoscaling != nil {
+			for _, ms := range machineSets {
+				if rMA.Name == ms.Name {
+					delete = false
+					break
+				}
+			}
 		}
+		if delete {
+			machineAutoscalersToDelete = append(machineAutoscalersToDelete, &remoteMachineAutoscalers.Items[i])
+		}
+	}
 
-		// Apply hive MachinePool taints to MachineSet MachineSpec.
-		ms.Spec.Template.Spec.Taints = pool.Spec.Taints
+	for _, ma := range machineAutoscalersToCreate {
+		logger.WithField("machineautoscaler", ma.Name).Info("creating machineautoscaler")
+		if err := remoteClusterAPIClient.Create(context.Background(), ma); err != nil {
+			logger.WithError(err).Error("unable to create machine autoscaler")
+			return err
+		}
 	}
 
-	return machineSets, nil
+	for _, ma := range machineAutoscalersToUpdate {
+		logger.WithField("machineautoscaler", ma.Name).Info("updating machineautoscaler")
+		if err := remoteClusterAPIClient.Update(context.Background(), ma); err != nil {
+			logger.WithError(err).Error("unable to update machine autoscaler")
+			return err
+		}
+	}
+
+	for _, ma := range machineAutoscalersToDelete {
+		logger.WithField("machineautoscaler", ma.Name).Info("deleting machineautoscaler")
+		if err := remoteClusterAPIClient.Delete(context.Background(), ma); err != nil {
+			logger.WithError(err).Error("unable to delete machine autoscaler")
+			return err
+		}
+	}
+
+	logger.Info("done reconciling machine autoscalers for cluster deployment")
+	return nil
+}
+
+func (r *ReconcileRemoteMachineSet) syncClusterAutoscaler(
+	pool *hivev1.MachinePool,
+	cd *hivev1.ClusterDeployment,
+	remoteClusterAPIClient client.Client,
+	logger log.FieldLogger,
+) error {
+	if pool.Spec.Autoscaling == nil {
+		return nil
+	}
+	remoteClusterAutoscalers := &autoscalingv1.ClusterAutoscalerList{}
+	tm := metav1.TypeMeta{}
+	tm.SetGroupVersionKind(autoscalingv1.SchemeGroupVersion.WithKind("ClusterAutoscaler"))
+	if err := remoteClusterAPIClient.List(
+		context.Background(),
+		remoteClusterAutoscalers,
+		client.UseListOptions(&client.ListOptions{Raw: &metav1.ListOptions{TypeMeta: tm}}),
+	); err != nil {
+		logger.WithError(err).Error("unable to fetch remote cluster autoscalers")
+		return err
+	}
+	logger.Infof("found %v remote cluster autoscalers", len(remoteClusterAutoscalers.Items))
```

Comment:
Should we warn if this is over 1?

Reply:
I don't know that it is Hive's job to warn about this. I am hesitant to add warning log events to Hive for things that do not affect Hive's behavior. Any ClusterAutoscaler that is not named "default" is (1) ignored by Hive and (2) ignored by the autoscaler.